platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/join/DefaultRoundDurabilityBuffer.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.event.preconsensus.join;
+
+import static com.swirlds.common.units.TimeUnit.UNIT_MICROSECONDS;
+import static com.swirlds.common.units.TimeUnit.UNIT_NANOSECONDS;
+import static com.swirlds.common.utility.CompareTo.isGreaterThan;
+import static com.swirlds.logging.legacy.LogMarker.EXCEPTION;
+
+import com.swirlds.base.time.Time;
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.common.metrics.RunningAverageMetric;
+import com.swirlds.common.utility.throttle.RateLimitedLogger;
+import com.swirlds.metrics.api.LongGauge;
+import com.swirlds.platform.event.preconsensus.PcesConfig;
+import com.swirlds.platform.internal.ConsensusRound;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * A default implementation of {@link RoundDurabilityBuffer}.
+ */
+public class DefaultRoundDurabilityBuffer implements RoundDurabilityBuffer {
+
+    private static final Logger logger = LogManager.getLogger(DefaultRoundDurabilityBuffer.class);
+
+    private long durableSequenceNumber = -1;
+    private final Queue<NotYetDurableRound> rounds = new LinkedList<>();
+
+    private final Time time;
+    private final Duration suspiciousRoundDuration;
+
+    private final RateLimitedLogger suspiciousRoundLogger;
+
+    private static final RunningAverageMetric.Config AVERAGE_ROUND_DURABILITY_DELAY_METRIC_CONFIG =
+            new RunningAverageMetric.Config("platform", "averageRoundDurabilityDelay")
+                    .withUnit("us")
+                    .withDescription(
+                            "The average delay between a round being eligible for handling and when the round's keystone event is guaranteed to be durable.");
+    private final RunningAverageMetric averageRoundDurabilityDelayMetric;
```

Comment:
I'm pretty curious to see what this value actually is

Reply:
Locally it was 3-10 milliseconds on average. I expect this could be wildly different on a real node with a real workload, but it's hard to say how different and in which direction.